src/test/javascript/spec/component/modeling-assessment/modeling-assessment.component.spec.ts

```diff
@@ -165,6 +181,60 @@ describe('ModelingAssessmentComponent', () => {
         expect(comp.feedbacks).toEqual(mockFeedbacks);
     });
 
+    it('should filter references by result feedbacks', () => {
+        expect(comp.referencedFeedbacks).toBeEmpty();
+        expect(comp.feedbacks).toBe(undefined);
+
+        comp.umlModel = mockModel;
+        comp.resultFeedbacks = mockFeedbacks;
+
+        expect(comp.referencedFeedbacks).toEqual([mockFeedbackWithReference, mockFeedbackInvalid]);
+        expect(comp.feedbacks).toBe(mockFeedbacks);
+    });
+
+    it('should calculate drop info', () => {
+        const spy = jest.spyOn(translatePipe, 'transform');
+        comp.umlModel = mockModel;
+        comp.resultFeedbacks = [mockFeedbackWithGradingInstruction];
+
+        expect(spy).toHaveBeenCalledWith('artemisApp.assessment.messages.removeAssessmentInstructionLink');
+        expect(spy).toHaveBeenCalledWith('artemisApp.exercise.assessmentInstruction');
+        expect(spy).toHaveBeenCalledWith('artemisApp.assessment.feedbackHint');
+        expect(mockModel.assessments[0].dropInfo.instruction).toBe(mockFeedbackWithGradingInstruction.gradingInstruction);
+    });
+
+    it('should update element counts', () => {
+        function getElementCounts(model: UMLModel): ModelElementCount[] {
+            // Not sure whether this is the correct logic to build OtherModelElementCounts.
+            return model.elements.map((el) => ({
+                elementId: el.id,
+                numberOfOtherElements: model.elements.length - 1,
+            }));
+        }
+
+        comp.umlModel = mockModel;
+        const elementCounts = getElementCounts(mockModel);
+        comp.elementCounts = elementCounts;
+
+        const spy = jest.spyOn(translatePipe, 'transform');
+
+        fixture.detectChanges();
+
+        elementCounts.forEach((elementCount) =>
+            expect(spy).toHaveBeenCalledWith('modelingAssessment.impactWarning', { affectedSubmissionsCount: elementCount.numberOfOtherElements }),
```

Comment:
We should expect a call count here as well

Reply:
Added the `toHaveBeenCalledTimes` assertion.